tests/kernel/fatal/src/main.c

```diff
@@ -235,15 +235,9 @@ void test_fatal(void)
 			(k_thread_entry_t)stack_thread2,
 			NULL, NULL, NULL, K_PRIO_PREEMPT(PRIORITY), 0,
 			K_NO_WAIT);
-#ifdef CONFIG_ARM
-	/* FIXME: See #7706 */
-	zassert_true(crash_reason == _NANO_ERR_STACK_CHK_FAIL ||
-		     crash_reason == _NANO_ERR_HW_EXCEPTION, NULL);
```

Comment:
We still need the workaround for the k64 because the nxp mpu throws a bus fault, not an mpu fault.